src/transformers/models/clap/configuration_clap.py

```diff
@@ -0,0 +1,453 @@
+        audio_config (`dict`, *optional*):
+            Dictionary of configuration options used to initialize [`ClapAudioConfig`].
+        projection_dim (`int`, *optional*, defaults to 512):
+            Dimentionality of text and audio projection layers.
+        logit_scale_init_value (`float`, *optional*, defaults to 2.6592):
+            The inital value of the *logit_scale* paramter. Default is used as per the original CLAP implementation.
+        fusion_num_hidden_layers (`int`, *optional*, defaults to 2):
+            Number of hidden layers in the fusion layer.
+        projection_dim (`int`, *optional*, defaults to 512):
+            Dimentionality of text and audio projection layers.
+        projection_hidden_act (`str`, *optional*, defaults to `"relu"`):
+            Activation function for the projection layers.
+        initializer_factor (`float`, *optional*, defaults to 1.0):
+            Factor to scale the initialization of the model weights.
+        kwargs (*optional*):
+            Dictionary of keyword arguments.
+
+    Example:
+
+    ```python
+    >>> from transformers import ClapConfig, ClapModel
+
+    >>> # Initializing a ClapConfig with laion-ai/base style configuration
+    >>> configuration = ClapConfig()
+
+    >>> # Initializing a ClapModel (with random weights) from the laion-ai/base style configuration
+    >>> model = ClapModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+
+    >>> # We can also initialize a ClapConfig from a ClapTextConfig and a ClapAudioConfig
+    >>> from transformers import ClapTextConfig, ClapAudioConfig
+
+    >>> # Initializing a ClapText and ClapAudioConfig configuration
+    >>> config_text = ClapTextConfig()
+    >>> config_audio = ClapAudioConfig()
+
+    >>> config = ClapConfig.from_text_audio_configs(config_text, config_audio)
+    ```"""
+
+    model_type = "clap"
+    is_composition = True
+
+    def __init__(
+        self,
+        text_config=None,
+        audio_config=None,
+        logit_scale_init_value=(1 / 0.07),
+        fusion_num_hidden_layers=2,
+        projection_dim=512,
+        projection_hidden_act="relu",
+        initializer_factor=1.0,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        if text_config is None:
+            text_config = {}
+            logger.info("text_config is None. Initializing the ClapTextConfig with default values.")
+
+        if audio_config is None:
+            audio_config = {}
+            logger.info("audio_config is None. initializing the ClapAudioConfig with default values.")
+
+        self.text_config = ClapTextConfig(**text_config)
+        self.audio_config = ClapAudioConfig(**audio_config)
+
+        self.text_config.fusion_num_hidden_layers = fusion_num_hidden_layers
+        self.audio_config.fusion_num_hidden_layers = fusion_num_hidden_layers
+
+        self.text_config.projection_dim = projection_dim
```

Comment:
Do we need to add `projection_dim` defaults `ClapTextConfig` and `ClapAudioConfig`? These attributes aren't present by default in these two configs, and are only set if we pass them when we instantiate. So currently, `projection_dim` and `projection_hidden_act` are only set if we create configs from `ClapConfig` or if we set them explicitly.

For reference, we add `projection_dim` by default in `CLIPTextConfig`: https://github.com/huggingface/transformers/blob/e3d832ff87c6ec997125deaa4f1b239db8f9e613/src/transformers/models/clip/configuration_clip.py#L99